### DISCUSSION TITLE
Backwards-compatible implementation of ToUrlSegment()

### DISCUSSION THREAD

**Discussion by JoseMarcenaro** [umbraco/Umbraco-CMS#11173] [Features and ideas]
Sometimes I have the impression that the DI dogma has been pushed a little bit too far in v9, making everyday tasks more complicated than what they should be.

An example of that is the v8 extension method `ToUrlSegment( IShortStringHelper shortStringHelper)`, in which the dependency injection version adds the flexibility of choosing the short string helper ... at the cost of forcing to do it every time and breaking the compatibility with v8 - even when in many cases the default implementation is Ok.

Note that this utility method is frequently used inside extension methods / static helpers, so getting a handler to the DI helper is not trivial.

I will submit a PR adding a default value for the parameter which uses the default implementation.

**Comment by JoseMarcenaro** [comment]
PR #11174 submitted

**Comment by greystate** [comment]
Ha - Was just about to complain loudly over this too — thanks for beating me to it 👍 😁
